### PR TITLE
Camera eyes have been lightly refactored  (among other things...)

### PR DESCRIPTION
**Pull request by tontyGH**
## About The Pull Request

* A generic /mob/eye/camera type has been made, containing everything needed to interface with a cameranet
* /mob/eye/ai_eye has been refactored into a generic /mob/eye/camera instance
* Advanced cameras no longer inherit from AI eyes, splitting off behaviour
* Camera code has been somewhat cleaned up
* Probably some more stuff I'm forgetting right now

## Big man Southport:
image

## Changelog

:cl:
code: made /proc/getviewsize() pure

refactor: mob/eye/ai_eye has been restructured, now inheriting from a generic mob/eye/camera type
refactor: advanced cameras and their subtypes are now mob/eye/camera/remote subtypes
code: the cameranet no longer expects the user to be an AI eye
code: remote camera eyes have had their initialization streamlined
code: remote cameras handle assigning and unassigning users by themselves now
code: remote cameras now use weakrefs instead of hard referencing owners and origins
code: also the sentient disease is_define was removed (we don't have those anymore)
fix: AI eyes no longer assign real names to themselves, fixing their orbit name
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
we went from `/mob/camera` to `/mob/eye` to `/mob/eye/camera`

real.

File: code/game/machinery/computer/camera_advanced.dm [hunk 1]
```diff
@@ -148,113 +151,51 @@
...
+		to_chat(user, span_warning("\The [src] flashes a bunch of never-ending errors on the display. Something is really wrong."))
+		return
+
+	var/camera_location
+	var/turf/myturf = get_turf(src)
+	var/consider_zlock = (!!length(z_lock))
+
+	if(!eyeobj.use_visibility)
+		if(consider_zlock && !(myturf.z in z_lock))
+			camera_location = locate(round(world.maxx/2), round(world.maxy/2), z_lock[1])
```
Comment (Ghommie) [inline code/game/machinery/computer/camera_advanced.dm:172]:
```suggestion
			camera_location = locate(round(world.maxx * 0.5), round(world.maxy * 0.5), z_lock[1])
```

File: code/modules/mob/eye/camera/camera.dm [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	var/final/list/datum/camerachunk/visibleCameraChunks = list()
+	/// NxN Range of a single camera chunk.
+	var/static_visibility_range = 16
+
+/mob/eye/camera/Initialize(mapload)
+	. = ..()
+	GLOB.camera_eyes += src
+
+/mob/eye/camera/Destroy()
+	for(var/V in visibleCameraChunks)
```
Comment (Ghommie) [inline code/modules/mob/eye/camera/camera.dm:25]: No single letter vars.

File: code/game/machinery/computer/camera_advanced.dm [hunk 3]
```diff
@@ -148,113 +151,51 @@
 	if(!QDELETED(current_user))
 		to_chat(user, span_warning("The console is already in use!"))
 		return
+
 	var/mob/living/L = user
```
Comment (Ghommie) [inline code/game/machinery/computer/camera_advanced.dm:155]: Ditto, if you would.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**